src/holisticai/robustness/dataset_shift/_accuracy_degradation_profile.py

```diff
@@ -0,0 +1,345 @@
+
+    This function ensures that the input arrays have matching lengths and that the input parameters
+    (baseline_accuracy, threshold_percentual, and above_percentual) are within their expected ranges.
+
+    Parameters:
+    ----------
+    X_test : np.ndarray
+        The test set features array. This array is expected to have the same length as `y_test` and `y_pred`.
+    y_test : np.ndarray
+        The true labels for the test set. Must have the same length as `X_test` and `y_pred`.
+    y_pred : np.ndarray
+        The predicted labels for the test set, corresponding to `X_test`. Must have the same length as `X_test` and `y_test`.
+    baseline_accuracy : float
+        The baseline accuracy of the model, expected to be a value between 0 and 1.
+    threshold_percentual : float
+        The percentage threshold to determine significant accuracy degradation. Must be between 0 and 1.
+    above_percentual : float
+        The minimum percentage of samples that must exceed the accuracy threshold to avoid triggering an accuracy degradation alert. Must be between 0 and 1.
+
+    Raises:
+    ------
+    ValueError
+        If any of the following conditions are met:
+        - The length of `X_test`, `y_test`, and `y_pred` do not match.
+        - `threshold_percentual` is not in the range (0, 1].
+        - `above_percentual` is not in the range (0, 1].
+        - `baseline_accuracy` is not in the range (0, 1].
+
+    Example:
+    -------
+    >>> _validate_inputs(
+    ...     np.array([1, 2, 3]),
+    ...     np.array([0, 1, 0]),
+    ...     np.array([0, 1, 1]),
+    ...     0.9,
+    ...     0.8,
+    ...     0.95,
+    ... )
+    No error raised, input is valid.
+
+    >>> _validate_inputs(
+    ...     np.array([1, 2, 3]), np.array([0, 1]), np.array([0, 1, 1]), 0.9, 0.8, 0.95
+    ... )
+    ValueError: X_test, y_test, and y_pred must have the same length.
+    """
+
+    if len(X_test) != len(y_test) or len(y_test) != len(y_pred):
+        raise ValueError("X_test, y_test, and y_pred must have the same length.")
+    if not (0 < threshold_percentual <= 1):
+        raise ValueError("threshold_percentual must be between 0 and 1.")
+    if not (0 < above_percentual <= 1):
+        raise ValueError("above_percentual must be between 0 and 1.")
+    if not (0 < baseline_accuracy <= 1):
+        raise ValueError("baseline_accuracy must be between 0 and 1.")
+
+
+def _calculate_accuracies(
+    X_test: np.ndarray, y_test: np.ndarray, y_pred: np.ndarray, n_neighbors: int, step_size: float
+) -> tuple[pd.DataFrame, list[float]]:
+    """
+    Calculate accuracies by iteratively reducing the test set size and evaluating accuracy.
+
+    This function evaluates the accuracy of predictions as the size of the test set is gradually reduced.
+    For each reduction, it calculates the accuracy over a subset of the nearest neighbors in the test set.
+
+    Parameters:
+    ----------
+    X_test : np.ndarray
```

Comment:
can be X_test a dataframe?
in tha case can be matrix-like for dataframe
for pd.Series is array-like

Reply:
Done.